### PR TITLE
[Makefile] Changes to enable help target and namespace creation during deployment if missing

### PR DESCRIPTION
**Pull request by nammn**
### All Submissions:

* [ ] Have you opened an Issue before filing this PR?
* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

## What problem does this PR solve?
This PR does the following:
- changes to the Makefile:
   - Refactoring (moving targets around, aligning comments for ease of help target scripting). 
   - Adds a help target (example below)
   - makes the `make deploy` target create and target the namespace if it does not exist. As of right now it would install it in the default namespace if it does not exist. Because of this `make undeploy` will fail as it targets a specific namespace. Both assuming targeting a namespace that does not exist
-  a small doc update, adding the missing second step 

Example `help` target
```
1.19.4 ~/projects/mongodb-kubernetes-operator nnguyen/doc-update*                                                                                                                                  03:46:19 PM
mongodb-kubernetes-operator ❯ make help  
Usage: make <OPTIONS> ... <TARGETS>

Available targets are:


Usage:
  make <target>

Development
  fmt                        Run go fmt against code
  vet                        Run go vet against code
  generate                   Generate code
  test                       Run unit tests
  manager                    Build manager binary
  run                        Run against the configured Kubernetes cluster in ~/.kube/config
  controller-gen             Download controller-gen locally if necessary
  helm                       Download helm locally if necessary
  install-prerequisites-macos  installs prerequisites for macos development

Installation/Uninstallation
  install                    Install CRDs into a cluster
  uninstall                  Uninstall CRDs from a cluster

Deployment
  deploy                     Deploy controller in the configured Kubernetes cluster in ~/.kube/config
  undeploy                   UnDeploy controller from the configured Kubernetes cluster in ~/.kube/config
  manifests                  Generate manifests e.g. CRD, RBAC etc.

E2E
  e2e-telepresence           Run e2e tests locally using go build while also setting up a proxy
  e2e-k8s                    Run e2e test by deploying test image in kubernetes.
  e2e                        Run e2e test locally. e.g. make e2e test=replica_set cleanup=true
  e2e-gh                     Trigger a Github Action of the given test

Image
  operator-image             Build and push the operator image
  e2e-image                  Build and push e2e test image
  agent-image                Build and push agent image
  readiness-probe-image      Build and push readiness probe image
  version-upgrade-post-start-hook-image  Build and push version upgrade post start hook image
  all-images                 create all required images
  help                       Show this help screen.

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -30,35 +30,61 @@ endif
...
+##@ Installation/Uninstallation
+
+install: manifests helm install-crd ## Install CRDs into a cluster
 
 install-crd:
 	kubectl apply -f config/crd/bases/mongodbcommunity.mongodb.com_mongodbcommunity.yaml
 
 install-chart:
-	$(HELM) upgrade --install $(STRING_SET_VALUES) $(RELEASE_NAME_HELM) $(HELM_CHART)
+	$(HELM) upgrade --install $(STRING_SET_VALUES) $(RELEASE_NAME_HELM) $(HELM_CHART) --namespace $(NAMESPACE) --create-namespace
```
Comment (nammn) [inline Makefile:84]: While running it locally I had the case that I setup a namespace that did not exist in my `config.json`.
With this the operator was installed in the `default` namespace, while a lot of other commands still assumed the one from `config.json`. For instance `helm undeploy` fails with the namespace target

File: Makefile [hunk 2]
```diff
@@ -74,50 +100,37 @@ uninstall-rbac:
...
-# Run go fmt against code
-fmt:
-	go fmt ./...
-
-# Run go vet against code
-vet:
-	go vet ./...
+##@ E2E
 
 # Run e2e tests locally using go build while also setting up a proxy in the shell to allow
```
Comment (nammn) [inline Makefile:117]: It did not feel correct to inline those 2 comments, hence I just left them where they are and added a new one below building on these

**Review by irajdeep — APPROVED** [review]
LGTM, nice cleanup

File: Makefile [hunk 3]
```diff
@@ -196,5 +184,9 @@ rm -rf $$TMP_DIR ;\
...
 endef
 
-install-prerequisites-macos:
-	scripts/dev/install_prerequisites.sh
+help: ## Show this help screen.
+	@echo 'Usage: make <OPTIONS> ... <TARGETS>'
+	@echo ''
+	@echo 'Available targets are:'
+	@echo ''
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z0-9_-]+:.*?##/ { printf "  \033[36m%-25s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
```
[the remark this answers is not in the document]
Comment (nammn) [reply Makefile]: added!
969957b..b64d728

**Pull request merged**